core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -154,6 +158,49 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
     }
   }
 
+  /** BigQuery interval syntax: INTERVAL int64 time_unit. */
+  @Override public void unparseSqlIntervalLiteral(
+          SqlWriter writer, SqlIntervalLiteral literal, int leftPrec, int rightPrec) {
+    SqlIntervalLiteral.IntervalValue interval =
+            (SqlIntervalLiteral.IntervalValue) literal.getValue();
```

Comment:
Calcite allows a pure string interval value, if BigQuery only allows `int64` format, we should check and throw for other non-integers values.

Reply:
Good idea. Added a `Long.parseLong` to validate if the value string is a long.

By doing so I have to change the test because seems like it usually the range time unit has a non long value. So I lose the test of checking range time unit.